optuna/visualization/_hypervolume_history.py

```diff
@@ -117,21 +117,21 @@ def _get_hypervolume_history_info(
                 hypervolume_values.append(hypervolume)
                 continue
 
+        if (values_normalized > minimization_reference_point).any():
+            hypervolume_values.append(hypervolume)
```

Comment:
[nit] What about defining `hypervolume_values` as `np.empty(len(complete_trials), dtype=float)` instead of using `hypervolume` and the list `hypervolume_values`?